diffkemp/llvm_ir/single_c_builder.py

```diff
@@ -13,29 +15,41 @@ class SingleCBuilder(SingleLlvmFinder):
     Extends the SingleLlvmFinder class by compiling C file to LLVM IR file.
     """
     def __init__(self, source_dir, c_file_name, clang="clang",
-                 clang_append=[]):
+                 clang_append=[], default_optim=True):
         """
         :param clang: clang compiler to be used
         :param clang_append: list of args to add when compiling
+        :param default_optim: use default optimalisations flags
+            and run LLVM IR simplification passes
         """
         llvm_file_name = os.path.splitext(c_file_name)[0] + ".ll"
         SingleLlvmFinder.__init__(self, source_dir, llvm_file_name)
 
         self.c_file_name = c_file_name
         self.clang = clang
         self.clang_append = clang_append
+        self.default_optim = default_optim
         self.initialize()
 
     def str(self):
         return "single_c_file"
 
     def initialize(self):
+        """Compiles the C file to LLVM IR file and runs passes"""
         # Using clang to compile c file to llvm file.
-        command = [self.clang, "-S", "-emit-llvm", "-g", self.c_file_name,
-                   "-o", self.llvm_file_name]
+        command = [self.clang, self.c_file_name, "-o", self.llvm_file_name]
+        # Note: extending with the default options must be before
+        # extending with the clang_append option
+        # so the optimisations flags can be redefined
+        # (it looks like clang uses last specified optimisaton flag).
```

Comment:
What about this:
```suggestion
        # Note: clang uses the last specified optimization level so
        # extending with the default options must be done before
        # extending with the clang_append option.
```
(probably better wrapped)